backends/dnf/pk-backend-dnf.c

```diff
@@ -3435,6 +3435,17 @@ pk_backend_upgrade_system_thread (PkBackendJob *job, GVariant *params, gpointer
 	/* set the installonly limit one higher than usual to avoid removing any kernels during system upgrades */
 	dnf_sack_set_installonly_limit (sack, dnf_context_get_installonly_limit (job_data->context) + 1);
 
+	/* reset libgit2 module when upgrading to F31: https://bugzilla.redhat.com/show_bug.cgi?id=1762751 */
+	if (g_strcmp0 (release_ver, "31") == 0) {
+		const gchar *reset_modules[] = { "libgit2", NULL };
+		g_debug ("resetting libgit2 module");
+		if (!dnf_context_reset_modules (job_data->context, sack, reset_modules, &error)) {
```

Comment:
can this happen if the libgit2 module isn't available? should ti be ignored instead?

Reply:
Ah yes, that's probably a good idea

Reply:
OK, updated the PR to not error out in case the reset fails